src/store/CurrentAccount/actions.ts

```diff
@@ -0,0 +1,48 @@
+import { CurrentAccountInterface } from '@/interfaces'
+import { SessionStorageKeys } from 'sbc-common-components/src/util/constants'
+import { sleep } from '@/utils'
+
+export default {
+  /**
+   * Gets account info and stores it in the CurrentAccount store.
+   * Among other things, this is how we find out if this is a staff account.
+   */
+  async loadCurrentAccount (context): Promise<any> {
+    return new Promise((resolve, reject) => {
+      context.dispatch('waitForCurrentAccount')
+        .then((currentAccount) => {
+          const accountInfo: CurrentAccountInterface = {
+            accountType: currentAccount.accountType,
+            id: currentAccount.id,
+            label: currentAccount.label,
+            type: currentAccount.type
+          }
+          context.commit('setAccountInformation', accountInfo)
+          resolve()
+        })
+        .catch((error) => {
+          reject(error)
+        })
+    })
+  },
+
+  /**
+   * Gets current account from object in session storage.
+   * Wait up to 5 sec for current account to be synced (typically by SbcHeader).
+   */
+  waitForCurrentAccount (): Promise<any> {
+    return new Promise((resolve, reject) => {
+      let account = null
+      for (let i = 0; i < 50; i++) {
+        const currentAccount = sessionStorage.getItem(SessionStorageKeys.CurrentAccount)
+        account = JSON.parse(currentAccount)
+        if (account) break
+        sleep(100)
+      }
+      if (account) {
+        return resolve(account)
+      }
+      return reject(new Error('Gave up waiting for CurrentAccount from Session'))
```

Comment:
@severinbeauvais I think you're right.  It's considered best practice to throw an error.  This still allows the calling logic to `catch()` and handle the error in a graceful way, but the errors can then be tracked.  I really appreciate your suggestion!!

Reply:
I think you don't have to throw an error -- anything passed in the reject call will be passed to the caller's catch handler.

Reply:
That's the way I've done it previously, but the linter is complaining if I don't put an error in inside reject().  The specific error from the linter reads, "expected the promise rejection reason to be an error".  I'm going to do some further reading.... always learning!

Reply:
I bet it'll work if you get rid of that promise code altogether:
- return the value after your async work
- throw an exception if there's a problem

Promises work well with "then" and "catch" blocks -- that's how most of our network services methods work.